pyispyb/core/routes/samples.py

```diff
@@ -128,6 +128,30 @@ def get_samples(
     )
 
 
+@router.get("/components", response_model=paginated(schema.Component))
+def get_components(
+    page: dict[str, int] = Depends(pagination),
+    proposal: str = Depends(filters.proposal),
+) -> Paged[models.BLSample]:
+    """Get a list of available components"""
+    return crud.get_components(
+        proposal=proposal,
+        **page,
+    )
+
+
+@router.get("/components/types", response_model=list[schema.ComponentType])
```

Comment:
Returning a list is technically not a valid REST response. It expects a dict. Could be solved by simple paginating this response